pinot-common/src/main/java/org/apache/pinot/common/helix/ExtraInstanceConfig.java

```diff
@@ -44,4 +45,31 @@ public String getTlsPort() {
   public void setTlsPort(String tlsPort) {
     _proxy.getRecord().setSimpleField(PinotInstanceConfigProperty.PINOT_TLS_PORT.toString(), tlsPort);
   }
+
+  /**
+   * Returns an instance URL from the InstanceConfig. Will set the appropriate protocol and port. Returns null
+   * if the URL cannot be constructed.
+   */
+  public String getComponentUrl() {
```

Comment:
the javadoc is not clear. is getComponentUrl returning the URL (host/port) for the component? if so. which one would it be for broker/server ? for example I think REST API calls to server should route to "adminPort" but not "port". 

should we give it a better javadoc / name?

Reply:
InstanceConfig is a class used by all the Pinot components (broker, server, controller, minion). So this util was supposed to serve the purpose of getting the REST API URL for any of those components. So I can update the method name/javadoc to be clearer.

Thanks for pointing out that the server should route to "adminPort" instead of "port". Is this the case only for calls to server?

Reply:
yeah so InstanceConfig host - port is sort of like a "helix" participant concept. where participate registers with helix of its service host/port. 
- for server there's admin, netty and grpc port. the main propose of pinot server is to talk to pinot-broker so the netty port is usually registered with helix (as the main service it provides). 
- if the idea is to provide a user facing endpoint, then the admin port is the one user will directly interact.

Reply:
i would suggest renaming it to getComponengRestResourceUrl() if the goal is to interact with user. and always use the REST API.